Modules/Devices/src/Devices.Domain/Entities/Identities/Identity.cs

```diff
@@ -84,6 +88,12 @@ public IdentityDeletionProcess StartDeletionProcessAsOwner(DeviceId asDevice)
         return deletionProcess;
     }
 
+    private void EnsureIdentityOwnsDevice(DeviceId currentDeviceId)
+    {
+        if (Devices.All(device => device.Id != currentDeviceId))
```

Comment:
Wow, this is quite some mental gymnastics... It works though. Maybe it should be simplified, I'd let @tnotheis decide.

Reply:
At this point I resolved all other comments. Waiting for Timo's decision on this.

Reply:
How about

```suggestion
        if (!Devices.Exists(device => device.Id == currentDeviceId))
```

Should be easier to understand. What do you think @NikolaVetnic?

Reply:
Absolutely.